tests/release/pipelines/release_to_github.go

```diff
@@ -0,0 +1,248 @@
+					Expect(tekton.HasPipelineRunSucceeded(buildPR)).To(BeTrue(), fmt.Sprintf("build pipelinerun %s/%s did not succeed", buildPR.GetNamespace(), buildPR.GetName()))
+					snapshot, err = devFw.AsKubeDeveloper.IntegrationController.GetSnapshot("", buildPR.Name, "", devNamespace)
+					if err != nil {
+						return err
+					}
+					return nil
+				}, releasecommon.BuildPipelineRunCompletionTimeout, releasecommon.DefaultInterval).Should(Succeed(), "timed out when waiting for build pipelinerun to be created")
+				//Expect(devFw.AsKubeDeveloper.HasController.WaitForComponentPipelineToBeFinished(component, "", devFw.AsKubeDeveloper.TektonController, &has.RetryOptions{Retries: 3, Always: true})).To(Succeed())
+			})
+			It("verifies the rhcs release pipelinerun is running and succeeds", func() {
+				releaseCR, err = devFw.AsKubeDeveloper.ReleaseController.GetRelease("", snapshot.Name, devNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+				Eventually(func() error {
+					releasePR, err = managedFw.AsKubeAdmin.ReleaseController.GetPipelineRunInNamespace(managedFw.UserNamespace, releaseCR.GetName(), releaseCR.GetNamespace())
+					if err != nil {
+						return err
+					}
+					Expect(err).ShouldNot(HaveOccurred())
+
+					if !releasePR.IsDone() {
+						return fmt.Errorf("release pipelinerun %s in namespace %s did not finish yet", releasePR.Name, releasePR.Namespace)
+					}
+					GinkgoWriter.Println("Release PR: ", releasePR.Name)
+					Expect(tekton.HasPipelineRunSucceeded(releasePR)).To(BeTrue(), fmt.Sprintf("release pipelinerun %s/%s did not succeed", releasePR.GetNamespace(), releasePR.GetName()))
+					return nil
+				}, releasecommon.ReleasePipelineRunCompletionTimeout, releasecommon.DefaultInterval).Should(Succeed(), "timed out when waiting for release pipelinerun to succeed")
+			})
+
+			It("verifies release CR completed and set succeeded.", func() {
+				Eventually(func() error {
+					releaseCR, err = devFw.AsKubeDeveloper.ReleaseController.GetRelease("", snapshot.Name, devNamespace)
+					if err != nil {
+					return err
+					}
+					GinkgoWriter.Println("Release CR: ", releaseCR.Name)
+					if !releaseCR.IsReleased() {
+						return fmt.Errorf("release %s/%s is not marked as finished yet", releaseCR.GetNamespace(), releaseCR.GetName())
+					}
+					return nil
+				}, releasecommon.ReleaseCreationTimeout, releasecommon.DefaultInterval).Should(Succeed())
+			})
+
+			It("verifies if the Release exists in github repo", func() {
+				trReleasePr, err := managedFw.AsKubeAdmin.TektonController.GetTaskRunStatus(managedFw.AsKubeAdmin.CommonController.KubeRest(), releasePR, "create-github-release")
+				Expect(err).NotTo(HaveOccurred())
+				trReleaseURL := trReleasePr.Status.TaskRunStatusFields.Results[0].Value.StringVal
+				Expect(trReleaseURL).To(Equal(rhcsReleaseURL))
+				Expect(gh.CheckIfReleaseExist(rhcsRepoOwner, "terraform-provider-rhcs", trReleaseURL)).To(BeTrue(), fmt.Sprintf("release %s doesn't exist", trReleaseURL))
```

Comment:
shouldn't we set up a repo or something just for this suite for a test github release?

Reply:
I cloned a [repo](https://github.com/terraform-redhat/terraform-provider-rhcs) to `redhat-appstudio-qe group` for the testing now. Do you mean we create a simple repo for test?

Reply:
I cloned a [repo](https://github.com/terraform-redhat/terraform-provider-rhcs) to redhat-appstudio-qe group for the testing now. Do you mean we create a simple repo for test?

Reply:
That is fine. If it is a repo in redhat-appstudio-qe with an appropriate name, I am good with it. I just don't want us stepping on the toes of whatever the terraform product is for our testing

Reply:
It will not affect terraform product. It will only release to the cloned repo.

Reply:
A new [repo](https://github.com/redhat-appstudio-qe/devfile-sample-go-basic/) is used.